frontend/simple/js/state.js

```diff
@@ -213,36 +214,36 @@ const actions = {
     Vue.events.$emit(hash, contractId, entry)
     Vue.events.$emit('eventHandled', contractId, entry)
   },
-
   // persisting the state
   async saveSettings (
     {state}: {state: Object}
   ) {
-    // TODO: encrypt these
-    const settings = {
-      position: state.position,
-      currentGroupId: state.currentGroupId,
-      contracts: Object.keys(state.contracts).map(contractId => ({
-        contractId,
-        type: state.contracts[contractId],
-        data: state[contractId]
-      }))
+    if (state.loggedIn) {
+      // TODO: encrypt these
+      const settings = {
+        position: state.position,
+        currentGroupId: state.currentGroupId,
+        contracts: Object.keys(state.contracts).map(contractId => ({
+          contractId,
+          ...state.contracts[ contractId ],
+          data: state[ contractId ]
+        }))
+      }
+      await db.saveSettings(state.loggedIn.name, settings)
```

Comment:
There is no such thing as `loggedIn.name`. Why did your tests not catch this? See if you can add tests to catch this?